AnnoDesigner/ViewModels/MainViewModel.cs

```diff
@@ -945,6 +946,46 @@ private void CanvasNormalize(object param)
             AnnoCanvas.Normalize(1);
         }
 
+        public ICommand MergeRoadsCommand { get; private set; }
+
+        /// <summary>
+        /// Filters all roads in current layout, finds largest groups of them and replaces them with merged variants.
+        /// </summary>
+        private void MergeRoads(object param)
+        {
+            var minX = AnnoCanvas.PlacedObjects.Where(p => p.WrappedAnnoObject.Road).Min(p => (int)p.Position.X);
+            var maxX = AnnoCanvas.PlacedObjects.Where(p => p.WrappedAnnoObject.Road).Max(p => (int)(p.Position.X + p.Size.Width));
+            var minY = AnnoCanvas.PlacedObjects.Where(p => p.WrappedAnnoObject.Road).Min(p => (int)p.Position.Y);
+            var maxY = AnnoCanvas.PlacedObjects.Where(p => p.WrappedAnnoObject.Road).Max(p => (int)(p.Position.Y + p.Size.Height));
+            var cells = Enumerable.Range(0, maxX - minX).Select(i => new LayoutObject[maxY - minY]).ToArray();
+            foreach (var item in AnnoCanvas.PlacedObjects.Where(p => p.WrappedAnnoObject.Road))
```

Comment:
Could we have braces around this for readability please? At least around the outer foreach - I can kind of see not doing it for the inner for loops.